src/data/set6/champions.ts

```diff
@@ -56,14 +56,13 @@ export default {
 	[ChampionKey.Kassadin]: {
 		cast: (elapsedMS, spell, champion) => {
 			const target = champion.target
-			const manaReave = champion.getSpellVariable(SpellKey.ManaReave) ?? 0
-			const castTime = champion.getCurrentSpell()?.castTime ?? 0
-			if (!target) { return console.log('No target', champion.name, champion.team) }
 			champion.queueProjectile(elapsedMS, spell, {
 				onCollision: () => {
-					if (!target.getBonusesFrom(SpellKey.ManaReave).length) {
-						target.bonuses.push([SpellKey.ManaReave, [[BonusKey.ManaReductionPercent, -100 * manaReave, elapsedMS + castTime]]]) 
-					}
+					const manaReave = champion.getSpellVariable(SpellKey.ManaReave) ?? 0
+					const duration = champion.getSpellVariable(SpellKey.Duration) ?? 0
+					const damageReduction = champion.getSpellVariable(SpellKey.DamageReduction) ?? 0
+					target?.bonuses.push([SpellKey.ManaReave, [[BonusKey.ManaReductionPercent, manaReave * -100]]])
```

Comment:
To test this spell I found it helpful to put a Blue Buff on Kassadin and have him fight 3* Galio. With these mana reaves stacking, Galio will never cast :)

Let's change this line to `target.setBonusesFor(ChampionKey.Kassadin, [BonusKey.ManaReductionPercent, manaReave * -100])`. Whenever any Kassadin hits a target with its spell, this will overwrite `ChampionKey.Kassadin` with the new manaReave value, which ensures this effect doesn't stack. (Keep in mind `ChampionKey.Kassadin` here is nothing but a label that controls how the passed bonuses get overwritten or not.)

Before this commit, you had `if (!target.getBonusesFrom(SpellKey.ManaReave).length) {` which is just another way of accomplishing the same thing (since `SpellKey.ManaReave` was the label key).